Sources/Misc/DangerousSettings.swift

```diff
@@ -45,14 +60,18 @@ import Foundation
      * If this is disabled, RevenueCat won't observe the StoreKit queue, and it will not sync any purchase
      * automatically.
      */
-    @objc public convenience init(autoSyncPurchases: Bool) {
-        self.init(autoSyncPurchases: autoSyncPurchases, internalSettings: .default)
+    @objc public convenience init(autoSyncPurchases: Bool = true, minimalImplementationOnly: Bool = false) {
+        self.init(autoSyncPurchases: autoSyncPurchases,
+                  minimalImplementationOnly: minimalImplementationOnly,
+                  internalSettings: .default)
+
     }
 
     /// Designated initializer
-    internal init(autoSyncPurchases: Bool, internalSettings: InternalSettings) {
+    internal init(autoSyncPurchases: Bool, minimalImplementationOnly: Bool, internalSettings: InternalSettings) {
```

Comment:
thought: we might want to split this into a few dangerous settings, making it much more modular. 
In particular, I'm thinking of: 
- one setting that disables logOut, and ideally also disables anonymous id creation. Basically it'd have to error out if you try to configure without appUserID. 
- one setting that disables getCustomerInfo, so that apps that do their own entitlements computation would be able to use that. 
- one or more settings to disable some other stuff, like attribution, subs attributes, trial eligibility

Reply:
Idea:
```swift
enum Feature {
  case attribution
  case authentication
  case entitlements
}

let disabledFeatures: Set<Feature> = []
```